### PR TITLE
Add Bosses to VGroids

### PR DESCRIPTION
**Pull request by ErhardSteinhauer**
## About the PR
Added the following bosses:
- Basalt VGroid: Mercenary Captain.
- Cave VGroid: Explorer Captain (ghost role). Added T3 spawns to the interior.
- Chromite VGroid: Ascended Cultist.
- Scrap VGroid: One Star Unit or Guardian Unit (random). Added drone spawns to the exterior.
- Snow VGroid: Xeno Queen (ghost role) and Leviathing.

## Why / Balance
Danger!

## How to test
1. Use addgamerule command to spawn VGroid, eg: `addgamerule BluespaceDungeonCave `
2. Take control over a boss and leave the grid, mob should become pacified and despawned after ~15 sec off-grid.

## Requirements

- [x] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [x] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes

**Changelog**
:cl: erhardsteinhauer
- tweak: Hostile faction bosses can now be encountered on VGroids.

### CODE REVIEW

File: Resources/Prototypes/_NF/Procedural/scrap_vgroid.yml [hunk 1]
```diff
@@ -144,6 +144,13 @@
       entity: SpawnMobRogueScapT1
       count: 25
       minGroupSize: 1
+      maxGroupSize: 2
```
Comment (dvir001) [inline Resources/Prototypes/_NF/Procedural/scrap_vgroid.yml:147]: MFW 50 Rogue

Comment (whatston3) [reply Resources/Prototypes/_NF/Procedural/scrap_vgroid.yml:147]: 65 mobs total with the drones below - might be a considerable problem given the limit of 128 AIs at once - we could increase it and see if it chugs.

File: Resources/Prototypes/_NF/Procedural/basalt_vgroid.yml [hunk 2]
```diff
@@ -242,6 +242,18 @@
...
       amount: 1
+  - !type:MobsDunGen
+    minCount: 1
+    maxCount: 3
+    groups:
+    - id: SpawnMobMercenaryT3
+      amount: 1
+  - !type:MobsDunGen
+    minCount: 1
+    maxCount: 3
```
Comment (whatston3) [inline Resources/Prototypes/_NF/Procedural/basalt_vgroid.yml:253]: Multiple bosses, or should this be 1-1?

**Comment by whatston3** [comment]
Largely seems fine giving it a read.  Will test briefly.

File: Resources/Prototypes/_NF/Procedural/scrap_vgroid.yml (hunk 1 above)
Comment (ErhardSteinhauer) [reply Resources/Prototypes/_NF/Procedural/scrap_vgroid.yml:147]: Kinda,  forgor to change numbers when copied the lines

File: Resources/Prototypes/_NF/Procedural/basalt_vgroid.yml (hunk 2 above)
Comment (ErhardSteinhauer) [reply Resources/Prototypes/_NF/Procedural/basalt_vgroid.yml:253]: I think ideally 1 boss per dungeon

**Comment by ErhardSteinhauer** [comment]
lowered the numbers

**Comment by whatston3** [comment]
New numbers make more sense, prior testing looked good.

**Review by whatston3 — APPROVED** [review]

**Pull request merged**